### DISCUSSION TITLE
Support aborting when() using AbortSignal

### DISCUSSION THREAD

**Discussion by deadbeef84** [mobxjs/mobx#3550] [Ideas]
I know the promise returned by `when()` can be aborted using `promise.cancel()`, but I think the new standard mechanism for aborting async operations is using AbortController / Signal.

I suggest adding a signal parameter to the options object:

```js
const abortController = new AbortController()
await when(() => ..., { signal: abortController.signal })
// abortController.abort() would make the promise reject
```

**Comment by mweststrate** [comment]
Sounds like a great idea. PR welcome :)

**Comment by deadbeef84** [comment]
https://github.com/mobxjs/mobx/pull/3551

**Comment by spaguette** [comment]
I was wondering if we could do the same with the MobX `flow` (e.g. in case you have some ongoing HTTP requests you'd like to cancel)?

**Reply by deadbeef84** [reply]
I haven't used MobX flow, but normally you would just pass a signal to your function, and forward it when calling async functions.
